### PR TITLE
feat: add vulnerability status

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR adds the new type `Status` and changes redhat and debian to use that. The detail is described [here](https://github.com/aquasecurity/trivy/issues/4763).

## Issue
https://github.com/aquasecurity/trivy/issues/4763

### CODE REVIEW

File: pkg/vulnsrc/debian/debian.go [hunk 1]
```diff
@@ -660,3 +660,19 @@ func compareVersions(v1, v2 string) (int, error) {
 
 	return ver1.Compare(ver2), nil
 }
+
+func newStatus(vulnID, s string) types.Status {
```
Comment (DmitriyLewen) [inline pkg/vulnsrc/debian/debian.go:664]: do you need `vulnID` here?

File: pkg/vulnsrc/redhat-oval/redhat-oval.go [hunk 2]
```diff
@@ -482,3 +485,20 @@ func severityFromImpact(sev string) types.Severity {
...
+		return types.StatusAffected
+	case "under investigation":
+		return types.StatusUnderInvestigation
+	case "will not fix":
+		return types.StatusWillNotFix
+	case "out of support scope":
+		return types.StatusEndOfLife
+	}
+	if s != "" {
+		panic(fmt.Sprintf("unknown status: %s", s))
```
Comment (DmitriyLewen) [inline pkg/vulnsrc/redhat-oval/redhat-oval.go:501]: Are you sure about `panic`?

Comment (knqyf263) [reply pkg/vulnsrc/redhat-oval/redhat-oval.go:501]: Oh, it was for testing...

File: pkg/vulnsrc/debian/debian.go (hunk 1 above)
Comment (knqyf263) [reply pkg/vulnsrc/debian/debian.go:664]: No, it was for testing...

**Review by DmitriyLewen — APPROVED** [review]

**Comment by knqyf263** [comment]
@jhebden-gl Just FYI: we changed state into status. We believe it will not break your system as it is a different field.

**Comment by jhebden-gl** [comment]
> @jhebden-gl Just FYI: we changed state into status. We believe it will not break your system as it is a different field.

Thank you so much for letting me know, I really appreciate it 🙇🏻

**Pull request merged**